--- a/address.go
+++ b/address.go
@@ -1,9 +1,10 @@
 package proton
 
 import (
 	"context"
 
 	"github.com/go-resty/resty/v2"
-	"golang.org/x/exp/slices"
+	"cmp"
+	"slices"
 )
 
 func (c *Client) GetAddresses(ctx context.Context) ([]Address, error) {
@@ -18,8 +19,8 @@ func (c *Client) GetAddresses(ctx context.Context) ([]Address, error) {
 	}); err != nil {
 		return nil, err
 	}
 
-	slices.SortFunc(res.Addresses, func(a, b Address) bool {
-		return a.Order < b.Order
+	slices.SortFunc(res.Addresses, func(a, b Address) int {
+		return cmp.Compare(a.Order, b.Order)
 	})
 
 	return res.Addresses, nil
